go-controller/pkg/factory/factory.go

```diff
@@ -1295,6 +1299,12 @@ func (wf *WatchFactory) addHandler(objType reflect.Type, namespace string, sel l
 	intInf.Lock()
 	defer intInf.Unlock()
 
+	// we are going to add a handler, we need to update the atomic signal that handlers exist now
+	// so that we do not miss events after we list current items.
```

Comment:
Nit: maybe change this to "we need to update the atomic signal that handlers exist now, before adding the actual handler, to ensure that we do not miss events while listing current items".

I'm not sure whether that sounds better or not, I'll leave it up to you.

Reply:
yeah +1 to updating this